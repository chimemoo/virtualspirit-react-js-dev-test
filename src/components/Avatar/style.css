.avatar {
  border: transparent 2px solid;
  border-radius: 50%;
}

.avatar img {
  border-radius: 50%;
  width: inherit;
  height: inherit;
}

.avatar.xs {
  width: 20px;
  height: 20px;
}

.avatar.sm {
  width: 30px;
  height: 30px;
}

.avatar.md {
  width: 40px;
  height: 40px;
}

.avatar.lg {
  width: 50px;
  height: 50px;
}