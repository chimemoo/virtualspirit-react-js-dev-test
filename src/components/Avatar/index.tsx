import { CSSProperties } from 'react'
import { getFirstTwoChars } from '../../utils'
import { AvatarSizeTypes } from '../types'
import './style.css'

interface AvatarProps {
  size: AvatarSizeTypes
  name: string
  url?: string
}

function Avatar({
  size,
  url,
  name,
}: AvatarProps) {
  if (!url) {
    return (
      <div className={`avatar placeholder ${size}`}>
        <span>{getFirstTwoChars(name)}</span>
      </div>
    )
  }

  return (
    <div className={`avatar ${size}`}>
      <img src={url} alt={name} />
    </div>
  )
}

export default Avatar;