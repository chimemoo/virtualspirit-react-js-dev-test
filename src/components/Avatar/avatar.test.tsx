import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { expect, test, describe } from 'vitest'
import Avatar from '.'
import avatar1 from '../../assets/avatar1.webp'


function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

describe('Avatar Component Test', () => {
  test('Avatar with Image Showed Correctly', () => {
    const component = renderer.create(
      <Avatar
        name="Budi Mulya"
        size="sm"
        url={avatar1}
      />
    )

    let tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  test('Avatar with Placeholder Showed Correctly', () => {
    const component = renderer.create(
      <Avatar
        name="Budi Mulya"
        size="sm"
      />
    )

    let tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  test('Avatar size expected', () => {
    render(<Avatar name="Budy Laode" size="lg" url={avatar1} />)

    const avatarClassName = screen.getByTestId('avatar-image').className

    expect(avatarClassName).contain('lg')
  })

  test('Avatar placeholder with 2 word name expected', () => {
    render(<Avatar name="Budy Laode" size="lg" />)

    const avatarPlaceholder = screen.getByTestId('avatar-placeholder').children[0].textContent

    expect(avatarPlaceholder).toBe("BL")
  })


  test('Avatar placeholder with 1 word name expected', () => {
    render(<Avatar name="Budy" size="lg" />)

    const avatarPlaceholder = screen.getByTestId('avatar-placeholder').children[0].textContent

    expect(avatarPlaceholder).toBe("B")
  })

  test('Avatar placeholder with more 2 word name expected', () => {
    render(<Avatar name="Budy Masasih Laode" size="lg" />)

    const avatarPlaceholder = screen.getByTestId('avatar-placeholder').children[0].textContent

    expect(avatarPlaceholder).toBe("BM")
  })
})