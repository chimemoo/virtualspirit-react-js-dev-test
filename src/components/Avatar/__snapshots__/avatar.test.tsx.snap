// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar Component Test > Avatar with Image Showed Correctly 1`] = `
<div
  className="avatar sm"
  data-testid="avatar-image"
>
  <img
    alt="Budi Mulya"
    src="/src/assets/avatar1.webp"
  />
</div>
`;

exports[`Avatar Component Test > Avatar with Placeholder Showed Correctly 1`] = `
<div
  className="avatar placeholder sm"
  data-testid="avatar-placeholder"
>
  <span>
    BM
  </span>
</div>
`;
